File: ssluzba konacno/src/main/java/rs/ac/uns/testdevelopment/ssluzba/pages/studenti/StudentsCreationPage.java
package rs.ac.uns.testdevelopment.ssluzba.pages.studenti;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import helpers.Utils;

public class StudentsCreationPage {
	private WebDriver driver;

	public StudentsCreationPage(WebDriver driver) {
		super();
		this.driver = driver;
	}

	public WebElement getModalDialog() {
		return Utils.waitForElementPresence(driver, By.className("modal-dialog"), 10);
	}

	public WebElement getModalTitle() {
		return Utils.waitForElementPresence(driver, By.id("myStudentiLabel"), 10);
	}

	public WebElement getIndex() {
		return Utils.waitForElementPresence(driver, By.name("indeks"), 10);
	}

	public void setIndex(String value) {
		WebElement el = this.getIndex();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getIme() {
		return Utils.waitForElementPresence(driver, By.name("ime"), 10);
	}

	public void setIme(String value) {
		WebElement el = this.getIme();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getPrezime() {
		return Utils.waitForElementPresence(driver, By.name("prezime"), 10);
	}

	public void setPrezime(String value) {
		WebElement el = this.getPrezime();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getGrad() {
		return Utils.waitForElementPresence(driver, By.name("grad"), 10);
	}

	public void setGrad(String value) {
		WebElement el = this.getGrad();
		el.clear();
		el.sendKeys(value);
	}

	public WebElement getCancelBtn() {
		return getModalDialog().findElement(By.className("btn-default"));
	}

	public WebElement getSaveBtn() {
		return getModalDialog().findElement(By.className("btn-primary"));
	}

	public void createStudent(String index, String ime, String prezime, String grad) {
		setIndex(index);
		setIme(ime);
		setPrezime(prezime);
		setGrad(grad);
		getSaveBtn().click();
	}

}

File: ssluzba konacno/src/test/javascript/e2e/pages/studenti/students-creation.page.js
// Prazan konstruktor
// Posto je "element" globalna funkcija, ne moramo da prosledimo nista.
var StudentCreationPage = function() {};
var utils = require('../utils.js');

// Opis polja i metoda
StudentCreationPage.prototype = Object.create({}, {
    
    modalDialog: {
        get: function() {
            return utils.waitForElementPresence(by.className('modal-dialog'), 10000);
        }
    },
    
    modalDialogTitle: {
        get: function() {
            return utils.waitForElementPresence(by.id('myStudentiLabel'), 10000).getText();
        }
    },
    
    index: {
        get: function() {
            return utils.waitForElementPresence(by.name('indeks'), 10000);
        },
        set: function(value) {
            return this.index.clear().sendKeys(value);
        }
    },
    
    ime: {
        get: function() {
            return utils.waitForElementPresence(by.name('ime'), 10000);
        },
        set: function(value) {
            return this.ime.clear().sendKeys(value);
        }
    },
    
    prezime: {
        get: function() {
            return utils.waitForElementPresence(by.name('prezime'), 10000);
        },
        set: function(value) {
            return this.prezime.clear().sendKeys(value);
        }
    },
    
    grad: {
        get: function() {
            return utils.waitForElementPresence(by.name('grad'), 10000);
        },
        set: function(value) {
            return this.grad.clear().sendKeys(value);
        }
    },
    
    cancelBtn: {
        get: function() {
            return this.modalDialog.element(by.className('btn-default'));
        }
    },
    
    saveBtn: {
        get: function() {
            return this.modalDialog.element(by.className('btn-primary'));
        }
    },
    
    createStudent: {
        value: function(index, ime, prezime, grad) {
            this.index = index;
            this.ime = ime;
            this.prezime = prezime;
            this.grad = grad;
            this.saveBtn.click();
        }
    }
});

// Export klase
module.exports = StudentCreationPage;